state_formatters/dp_formatters.py

```diff
@@ -1252,3 +1255,56 @@ def dff_command_selector_skill_formatter(dialog: Dict) -> List[Dict]:
     batches[-1]["dialog_batch"][-1]["called_intents"] = called_intents
     batches[-1]["dialog_batch"][-1]["dialog_id"] = dialog.get("dialog_id", "unknown")
     return batches
+
+
+def user_emotion_bot_mood_formatter(dialog: Dict) -> List[Dict]:
+    # comment out the next two lines to save dialog data to see its contents
+    # with open('test_formatters.json', 'w', encoding='utf-8') as f:
+    #     json.dump(dialog, f, ensure_ascii=False, indent=4)
+    sentences = [dialog["human_utterances"][-1]["text"]]
+    annotated_utterances = [dialog["human_utterances"][-1]]
+
+    if len(dialog["bot_utterances"]) > 1:
+        bot_mood = (
+            dialog["bot_utterances"][-2]["annotations"]
+            .get("bot_emotion_classifier", {})
+            .get("bot_mood", [0.75, 0.25, 0.44])
+        )
+    else:
+        bot_mood = [0.75, 0.25, 0.44]
+    bot_mood_list = len(sentences) * [bot_mood]
+
+    return [
+        {
+            "sentences": sentences,
+            "annotated_utterances": annotated_utterances,
+            "bot_mood": bot_mood_list,
+        }
+    ]
+
+
+def bot_mood_emotion_formatter(dialog: Dict) -> List[Dict]:
+    # comment out the next two lines to save dialog data to see its contents
+    # with open('test_formatters.json', 'w', encoding='utf-8') as f:
+    #     json.dump(dialog, f, ensure_ascii=False, indent=4)
+    hypotheses = dialog["human_utterances"][-1]["hypotheses"]
+    hypots = [h["text"] for h in hypotheses]
+
+    bot_mood_label = (
+        dialog["bot_utterances"][-1]["annotations"].get("bot_emotion_classifier", {}).get("bot_mood_label", "happy")
```

Comment:
what if this fucntion will be triggered on the 1st step of the dialog? When no bot uttrs

Reply:
added handling of this situation